aptos-move/framework/aptos-framework/doc/object.md

```diff
@@ -0,0 +1,1239 @@
+<b>use</b> <a href="event.md#0x1_event">0x1::event</a>;
+<b>use</b> <a href="../../aptos-stdlib/doc/from_bcs.md#0x1_from_bcs">0x1::from_bcs</a>;
+<b>use</b> <a href="guid.md#0x1_guid">0x1::guid</a>;
+<b>use</b> <a href="../../aptos-stdlib/doc/hash.md#0x1_hash">0x1::hash</a>;
+<b>use</b> <a href="../../aptos-stdlib/../move-stdlib/doc/signer.md#0x1_signer">0x1::signer</a>;
+<b>use</b> <a href="../../aptos-stdlib/../move-stdlib/doc/vector.md#0x1_vector">0x1::vector</a>;
+</code></pre>
+
+
+
+<a name="0x1_object_Object"></a>
+
+## Resource `Object`
+
+
+
+<pre><code><b>struct</b> <a href="object.md#0x1_object_Object">Object</a> <b>has</b> key
+</code></pre>
+
+
+
+<details>
+<summary>Fields</summary>
+
+
+<dl>
+<dt>
+<code>guid_creation_num: u64</code>
+</dt>
+<dd>
+
+</dd>
+<dt>
+<code>owner: <b>address</b></code>
+</dt>
+<dd>
+
+</dd>
+<dt>
+<code>allow_ungated_transfer: bool</code>
+</dt>
+<dd>
+
+</dd>
+<dt>
+<code>transfer_events: <a href="event.md#0x1_event_EventHandle">event::EventHandle</a>&lt;<a href="object.md#0x1_object_TransferEvent">object::TransferEvent</a>&gt;</code>
+</dt>
+<dd>
+
+</dd>
+</dl>
+
+
+</details>
+
+<a name="0x1_object_ObjectGroup"></a>
+
+## Struct `ObjectGroup`
+
+
+
+<pre><code><b>struct</b> <a href="object.md#0x1_object_ObjectGroup">ObjectGroup</a>
+</code></pre>
+
+
+
+<details>
+<summary>Fields</summary>
+
+
+<dl>
+<dt>
+<code>dummy_field: bool</code>
+</dt>
+<dd>
+
+</dd>
+</dl>
+
+
+</details>
+
+<a name="0x1_object_ObjectId"></a>
+
+## Struct `ObjectId`
+
+Type safe way of designate an object as at this address
+
+
+<pre><code><b>struct</b> <a href="object.md#0x1_object_ObjectId">ObjectId</a> <b>has</b> <b>copy</b>, drop, store
+</code></pre>
+
+
+
+<details>
+<summary>Fields</summary>
+
+
+<dl>
+<dt>
+<code>inner: <b>address</b></code>
+</dt>
+<dd>
+
+</dd>
+</dl>
+
+
+</details>
+
+<a name="0x1_object_CreatorRef"></a>
+
+## Struct `CreatorRef`
+
+This is a one time ability given to the creator to configure the object as necessary
+
+
+<pre><code><b>struct</b> <a href="object.md#0x1_object_CreatorRef">CreatorRef</a> <b>has</b> drop
+</code></pre>
+
+
+
+<details>
+<summary>Fields</summary>
+
+
+<dl>
+<dt>
+<code>self: <a href="object.md#0x1_object_ObjectId">object::ObjectId</a></code>
+</dt>
+<dd>
+
+</dd>
+<dt>
+<code>can_delete: bool</code>
+</dt>
+<dd>
+
+</dd>
+</dl>
+
+
+</details>
+
+<a name="0x1_object_DeleteRef"></a>
+
+## Struct `DeleteRef`
+
+The owner of this ability can delete an object
+
+
+<pre><code><b>struct</b> <a href="object.md#0x1_object_DeleteRef">DeleteRef</a> <b>has</b> drop, store
+</code></pre>
+
+
+
+<details>
+<summary>Fields</summary>
+
+
+<dl>
+<dt>
+<code>self: <a href="object.md#0x1_object_ObjectId">object::ObjectId</a></code>
+</dt>
+<dd>
+
+</dd>
+</dl>
+
+
+</details>
+
+<a name="0x1_object_ExtendRef"></a>
+
+## Struct `ExtendRef`
+
+Access to this ref allows for creation of the signer to add new resources
```

Comment:
What is the Ref that allows for the deletion of existing resources within an object?